apps/schema2/src/components/Yuml.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <div v-if="schema.tables">
+    <InputCheckbox
+      v-model="showAttributes"
+      :defaultValue="showAttributes"
+      :options="['attributes', 'external']"
+    />
+    <div style="text-align: center" class="overflow-auto">
+      <Spinner v-if="loadingYuml" />
+      <img
+        :class="imgFullscreen ? '' : 'img-fluid'"
+        @click="imgFullscreen = !imgFullscreen"
+        v-else
+        :key="JSON.stringify(showAttributes)"
+        :src="yuml"
+        alt="Small"
+        style="max-height: 100%"
+        @load="loadingYuml = false"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+import { Spinner, InputCheckbox } from "molgenis-components";
+
+export default {
+  components: {
+    Spinner,
+    InputCheckbox,
+  },
+  props: {
+    schema: Object,
+  },
+  data() {
+    return {
+      loadingYuml: false,
+      imgFullscreen: false,
+      showAttributes: [],
+    };
+  },
+  computed: {
+    tables() {
+      return this.schema.tables;
+    },
+    yuml() {
+      if (!this.tables || this.tables.length === 0) return "";
+      this.loadingYuml = true;
+      let res = "https://yuml.me/diagram/plain;dir:bt/class/";
+      // classes
+      this.tables
+        .filter(
+          (t) => !t.externalSchema || this.showAttributes.includes("external")
+        )
+        .forEach((table) => {
+          res += `[${table.name}`;
+
+          if (
+            Array.isArray(table.columns) &&
+            this.showAttributes.includes("attributes")
+          ) {
```

Comment:
lots of nested if and else and loops, at least is needs a test , suggest breaking up in names (inner ) functions